src/util/misc.ts

```diff
@@ -424,56 +425,47 @@ export function cleanupAxiosError(err: AxiosError) {
   return { name: err.name, message: err.message, url: err.config?.url, status: err.response?.status, responseType: err.config?.responseType, data: err.response?.data }
 }
 
-export async function readFileOrUrlByLine(resourcePath: string): Promise<string[]> {
-  if (resourcePath.includes('http') && !resourcePath.includes(',')) {
-    const fileName = resourcePath.split('/').slice(-1)[0]
-    const { data: contentStream } = await axios.get(resourcePath, {
-      headers: {
-        accept: 'application/octet-stream',
-        'cache-control': 'public, max-stale=86400',
-        'accept-encoding': 'gzip, deflate',
-        'user-agent': config.userAgent,
-      },
-      responseType: 'stream',
-      timeout: config.defaults.requestTimeout,
-      method: 'GET',
-    })
-    resourcePath = path.join(await getTmpDirectory(), fileName)
-    const writeStream = fs.createWriteStream(resourcePath)
-    await new Promise((resolve, reject) => {
-      contentStream
-        .pipe(writeStream)
-        .on('error', (err: any) => reject(err))
-        .on('close', resolve)
-    })
-  }
-
-  if (!fs.existsSync(resourcePath)) {
-    const list = await Promise.all(
-      resourcePath
-        .split(',')
-        .filter((n) => n)
-        .map(async (part) => {
-          const path = part.trim()
-          if (path.includes('http') || fs.existsSync(path)) {
-            return readFileOrUrlByLine(path)
+async function downloadListByUrl(url: string): Promise<string> {
+  const fileName = url.split('/').slice(-1)[0]
+  const { data: contentStream } = await axios.get(url, defaultStreamRequestOptions)
+  const filePath = path.join(await getTmpDirectory(), fileName)
+  const writeStream = fs.createWriteStream(filePath)
+  await new Promise((resolve, reject) => {
+    contentStream
+      .pipe(writeStream)
+      .on('error', (err: any) => reject(err))
+      .on('close', resolve)
+  })
+  return filePath
+}
+
+export async function extractArticleList(articleList: string): Promise<string[]> {
+  const list = await Promise.all(
+    articleList
+      .split(',')
+      .filter((n) => n)
+      .map(async (part) => {
+        let item: string | string[] = part.trim()
+        if (item.indexOf('http') === 0) {
+          try {
+            const url = new URL(item)
+            item = await downloadListByUrl(url.toString())
+          } catch (e) {
+            // URL is not valid continue processing
```

Comment:
Why a silent exception here? If a list can not be downloaded it should stops in a handled manner?